loader.rb

```diff
@@ -9,15 +9,56 @@
 
 require "bundler/setup"
 Bundler.setup
-require "zeitwerk"
-Loader = Zeitwerk::Loader.new
-Loader.push_dir("#{__dir__}/")
-Loader.push_dir("#{__dir__}/model")
-Loader.push_dir("#{__dir__}/lib")
-Loader.ignore("#{__dir__}/routes")
-Loader.ignore("#{__dir__}/migrate")
-Loader.ignore("#{__dir__}/spec")
-Loader.ignore("#{__dir__}/model.rb")
-Loader.inflector.inflect("db" => "DB")
-Loader.enable_reloading
-Loader.setup
+
+require_relative "./lib/casting_config_helpers"
+require_relative "./config"
+
+unless defined?(Unreloader)
+  require "rack/unreloader"
+  Unreloader = Rack::Unreloader.new(
+    reload: Config.development?,
+    autoload: !Config.production?
+  )
```

Comment:
`config.ru` overwrites the global `Unreloader` variable. If we run rackup, app uses second `Unreloader` instance. Should we combine them? Or is it intended?

```ruby
require_relative "loader"

require "rack/unreloader"
Unreloader = Rack::Unreloader.new(subclasses: %w[Roda Sequel::Model], logger: logger, reload: dev) { Clover }
require_relative "model"
Unreloader.require("clover.rb") { "Clover" }
```

https://github.com/ubicloud/ubicloud/blob/f9ed2cdf7e2fcd9f1d27ccde5679f12220829439/config.ru#L13

Reply:
I went ahead and deleted all other global Unreloader bindings. There are some touchy aspects to this, for example, teaching the global Unreloader object about the Rack app so that auto-reloading in a web context works, and also that I skipped the `subclasses:` filter, hoping that the fact that we rely exclusively on Autoload (and thus itemize the constants associated with a file) is enough.